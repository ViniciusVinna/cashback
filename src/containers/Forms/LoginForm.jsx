import React from 'react';
import { useDispatch } from 'react-redux';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers';

import { getUser } from 'actions';

import { loginSchema } from 'modules/form-validations';

import { Input, SubmitField } from 'components/Input';

import StyledForm from './Forms.styled';

const { Error, Form, Label } = StyledForm;

export const LoginForm = () => {
  const { errors, handleSubmit, register } = useForm({
    resolver: yupResolver(loginSchema),
  });
  const dispatch = useDispatch();

  const onSubmitHandler = (data) => {
    dispatch(getUser(data));
  };

  return (
    <Form onSubmit={handleSubmit(onSubmitHandler)} data-testid="login-form">
      <Label htmlFor="email">E-mail</Label>
      <Input
        hasError={!!errors.email}
        name="email"
        placeholder="digite seu email..."
        ref={register}
      />
      {!!errors.email && (<Error>{errors?.email?.message}</Error>)}

      <Label htmlFor="password">Senha</Label>
      <Input
        hasError={!!errors.password}
        name="password"
        placeholder="digite sua senha..."
        ref={register({ required: true })}
        type="password"
      />
      {!!errors.password && (<Error>{errors?.password?.message}</Error>)}

      <SubmitField type="submit">
        Entrar
      </SubmitField>
    </Form>
  );
};
