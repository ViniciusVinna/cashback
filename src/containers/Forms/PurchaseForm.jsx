import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers';
import { useDispatch, useSelector } from 'react-redux';

import { createPurchase } from 'actions';

import { clearNumber, formatCurrency, purchaseSchema } from 'modules';

import Button from 'components/Button';
import { Input } from 'components/Input';

import StyledForm from './Forms.styled';

const { Error, Form, Label, Money } = StyledForm;

export const PurchaseForm = () => {
  const [purchaseValue, setPurchaseValue] = useState('');
  const { create: { status } } = useSelector(state => state.purchases);
  const dispatch = useDispatch();
  const { errors, handleSubmit, register } = useForm({
    resolver: yupResolver(purchaseSchema),
  });

  const handleValue = (e) => {
    const { value } = e.target;

    /* istanbul ignore else */
    if (value) {
      setPurchaseValue(clearNumber(value));

      return;
    }

    setPurchaseValue('');
  };

  const onSubmitHandler = (data) => {
    const purchasedata = { ...data, value: clearNumber(data.value) };

    dispatch(createPurchase(purchasedata));
  };

  return (
    <Form
      onSubmit={handleSubmit(onSubmitHandler)}
      noBorder
      noMargin
      data-testid="purchase-form"
    >
      <Label htmlFor="value">Valor</Label>
      <Money
        name="value"
        placeholder="00,00"
        onChange={handleValue}
        autoFocus
        ref={register({ required: true })}
        type="tel"
        value={formatCurrency(purchaseValue)}
      />
      {!!errors.value && (<Error>{errors?.value?.message}</Error>)}

      <Label htmlFor="code">CÃ³digo</Label>
      <Input
        hasError={!!errors.code}
        name="code"
        placeholder="000000000000"
        ref={register({ required: true })}
        type="text"
      />
      {!!errors.code && (<Error>{errors?.code?.message}</Error>)}

      <Label htmlFor="date">Data</Label>
      <Input
        hasError={!!errors.date}
        name="date"
        type="date"
        placeholder="00/00/0000"
        ref={register({ required: true })}
      />
      {!!errors.date && (<Error>{errors?.date?.message}</Error>)}

      <Button type="submit" variation="primary" status={status}>
        Finalizar Cadastro
      </Button>
    </Form>
  );
};
