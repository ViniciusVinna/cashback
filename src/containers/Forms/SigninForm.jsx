import React from 'react';
import { useForm } from 'react-hook-form';

import Button from 'components/Button';
import { Input } from 'components/Input';

import StyledForm from './Forms.styled';

const { Form, Label } = StyledForm;

export const SigninForm = () => {
  const { register, handleSubmit, errors } = useForm();
  const onSubmit = data => console.log(data);

  return (
    <Form onSubmit={handleSubmit(onSubmit)} noBorder={true}>
      <Label htmlFor="username">Nome Completo</Label>
      <Input
        name="username"
        type="text"
        placeholder="digite seu nome..."
        ref={register}
        data-testid="signin-form"
      />

      <Label htmlFor="cpf">CPF</Label>
      <Input
        name="cpf"
        type="tel"
        placeholder="000.000.000-00"
        ref={register}
      />

      <Label htmlFor="email">E-mail</Label>
      <Input
        name="email"
        placeholder="digite seu email..."
        ref={register}
      />

      <Label htmlFor="password">Senha</Label>
      <Input
        name="password"
        placeholder="digite sua senha..."
        ref={register({ required: true })}
        type="password"
      />
      {errors.exampleRequired && <span>This field is required</span>}

      <Button type="submit" variation="secondary">
        Finalizar Cadastro
      </Button>
    </Form>
  );
};
