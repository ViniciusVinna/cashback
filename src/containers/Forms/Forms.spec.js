import React from 'react';
import { LoginForm } from './LoginForm';
import { PurchaseForm } from './PurchaseForm';
import { SigninForm } from './SigninForm';

describe('Forms', () => {
  describe('LoginForm', () => {
    it('should render properly', () => {
      const { container } = render(<LoginForm />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('PurchaseForm', () => {
    it('should render properly', () => {
      const { container } = render(<PurchaseForm />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('SigninForm', () => {
    it('should render properly', () => {
      const { container } = render(<SigninForm />);
      expect(container).toMatchSnapshot();
    });
  });
});
