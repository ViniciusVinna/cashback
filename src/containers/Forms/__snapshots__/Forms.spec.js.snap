// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms LoginForm should render properly 1`] = `
<div>
  <form
    class="css-1ywmfgl"
    data-testid="login-form"
  >
    <label
      class="css-13tv2p4"
      for="email"
    >
      E-mail
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
        />
        <polyline
          points="22,6 12,13 2,6"
        />
      </svg>
      <input
        class="css-309ol0"
        name="email"
        placeholder="digite seu email..."
      />
    </div>
    <label
      class="css-13tv2p4"
      for="password"
    >
      Senha
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="11"
          rx="2"
          ry="2"
          width="18"
          x="3"
          y="11"
        />
        <path
          d="M7 11V7a5 5 0 0 1 10 0v4"
        />
      </svg>
      <input
        class="css-309ol0"
        name="password"
        placeholder="digite sua senha..."
        type="password"
      />
    </div>
    <button
      class="css-tdfxln"
      type="submit"
    >
      Entrar
      <canvas
        height="0"
        style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        width="0"
      />
    </button>
  </form>
</div>
`;

exports[`Forms PurchaseForm should render properly 1`] = `
<div>
  <form
    class="css-105uino"
    data-testid="purchase-form"
  >
    <label
      class="css-13tv2p4"
      for="code"
    >
      CÃ³digo
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
        />
        <line
          x1="7"
          x2="7"
          y1="7"
          y2="7"
        />
      </svg>
      <input
        class="css-309ol0"
        name="code"
        placeholder="000000000000"
        type="text"
      />
    </div>
    <label
      class="css-13tv2p4"
      for="value"
    >
      Valor
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="1"
          y2="23"
        />
        <path
          d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
        />
      </svg>
      <input
        class="css-309ol0"
        name="value"
        placeholder="R$ 00,00"
        type="tel"
      />
    </div>
    <label
      class="css-13tv2p4"
      for="date"
    >
      Data
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="18"
          rx="2"
          ry="2"
          width="18"
          x="3"
          y="4"
        />
        <line
          x1="16"
          x2="16"
          y1="2"
          y2="6"
        />
        <line
          x1="8"
          x2="8"
          y1="2"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="10"
          y2="10"
        />
      </svg>
      <input
        class="css-309ol0"
        name="date"
        placeholder="00/00/0000"
        type="date"
      />
    </div>
    <button
      class="css-tdfxln"
      data-test-id="button"
      type="submit"
    >
      Finalizar Cadastro
      <canvas
        height="0"
        style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        width="0"
      />
    </button>
  </form>
</div>
`;

exports[`Forms SigninForm should render properly 1`] = `
<div>
  <form
    class="css-3mxqed"
  >
    <label
      class="css-13tv2p4"
      for="username"
    >
      Nome Completo
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
        />
        <circle
          cx="12"
          cy="7"
          r="4"
        />
      </svg>
      <input
        class="css-309ol0"
        data-testid="signin-form"
        name="username"
        placeholder="digite seu nome..."
        type="text"
      />
    </div>
    <label
      class="css-13tv2p4"
      for="cpf"
    >
      CPF
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.715,12c1.151,0,2-0.849,2-2s-0.849-2-2-2s-2,0.849-2,2S8.563,12,9.715,12z"
        />
        <path
          d="M20,4H4C2.897,4,2,4.841,2,5.875v12.25C2,19.159,2.897,20,4,20h16c1.103,0,2-0.841,2-1.875V5.875C22,4.841,21.103,4,20,4z M20,18L4,17.989V6l16,0.011V18z"
        />
        <path
          d="M14 9H18V11H14zM15 13H18V15H15zM13.43 15.536c0-1.374-1.676-2.786-3.715-2.786S6 14.162 6 15.536V16h7.43V15.536z"
        />
      </svg>
      <input
        class="css-309ol0"
        name="cpf"
        placeholder="000.000.000-00"
        type="tel"
      />
    </div>
    <label
      class="css-13tv2p4"
      for="email"
    >
      E-mail
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
        />
        <polyline
          points="22,6 12,13 2,6"
        />
      </svg>
      <input
        class="css-309ol0"
        name="email"
        placeholder="digite seu email..."
      />
    </div>
    <label
      class="css-13tv2p4"
      for="password"
    >
      Senha
    </label>
    <div
      class="css-bhvw3g"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="11"
          rx="2"
          ry="2"
          width="18"
          x="3"
          y="11"
        />
        <path
          d="M7 11V7a5 5 0 0 1 10 0v4"
        />
      </svg>
      <input
        class="css-309ol0"
        name="password"
        placeholder="digite sua senha..."
        type="password"
      />
    </div>
    <button
      class="css-exvprd"
      data-test-id="button"
      type="submit"
    >
      Finalizar Cadastro
      <canvas
        height="0"
        style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        width="0"
      />
    </button>
  </form>
</div>
`;
